host-controller/src/main/java/org/jboss/as/host/controller/DomainModelControllerService.java

```diff
@@ -1235,14 +1235,12 @@ private static AbstractVaultReader loadVaultReaderService() {
         final Iterator<AbstractVaultReader> it = serviceLoader.iterator();
         // TODO WFCORE-114 get rid of catching/suppressing errors once we have a complete impl in WFCORE
         ServiceConfigurationError sce = null;
-        while (it.hasNext()) {
-            try {
+        try {
+            while (it.hasNext()) {
                 return it.next();
-            } catch (ServiceConfigurationError e) {
-                if (sce == null) {
-                    sce = e;
-                }
             }
+        } catch (ServiceConfigurationError e) {
+            sce = e;
```

Comment:
Was there a reason to change this? It looks like previously it would only log the first error and now it will log the last error.